lib/options_validator.js

```diff
@@ -138,4 +175,78 @@ function invalidate(validationLevel, message, logger) {
   }
 }
 
-module.exports = { createValidationFunction };
+/**
+ * This is a class to build an operation by checking its arity and validating its options.
+ */
+class OperationBuilder {
+  constructor() {
+    this.requiredArity = -1;
+    this.validationSchema = {};
+  }
+}
+
+/**
+ * Build an OperationBuilder given an operation's required arity.
+ *
+ * @method
+ * @param {number} requiredArity The required arity of the operation to build.
+ * @return {OperationBuilder} An operation builder that will validate options.
+ */
+function arity(requiredArity) {
+  const operationBuilder = new OperationBuilder(requiredArity);
+  return operationBuilder.arity(requiredArity);
+}
+
+OperationBuilder.prototype.arity = function(requiredArity) {
```

Comment:
Is there any particular reason you defined these on the prototype, rather than just including them in the class definition above?

Reply:
Nope, I'll go ahead and include them in the class definition.